app/helpers/camaleon_cms/uploader_helper.rb

```diff
@@ -1,5 +1,17 @@
+# frozen_string_literal: true
+
 module CamaleonCms
   module UploaderHelper
+    UNSAFE_EXPRESSIONS = %w[script prompt eval url \%(css)s alert src= href= data="http redirect].freeze
```

Comment:
What do you think about this?
```ruby
def scan_file_for_malicious_content(file_path)
  # Read the file content
  file_content = File.read(file_path)

  # Check for suspicious patterns
  suspicious_patterns = [
    /<script[\s>]/i,  # Script tags
    /on\w+\s*=/i,     # Inline event handlers like onload, onclick, etc.
    /javascript:/i,   # JavaScript in href/src attributes
    /<iframe[\s>]/i,  # Iframes
    /<object[\s>]/i,  # Object tags
    /<embed[\s>]/i,   # Embed tags
    /<base[\s>]/i,    # Base tags (can be used to manipulate URLs)
    /data:/i          # data: URLs (which can include scripts)
  ]

  suspicious_patterns.each do |pattern|
    if file_content =~ pattern
      puts "Potentially malicious content found: #{pattern.inspect}"
      return true
    end
  end
end
```

Reply:
Yeah, brilliant, thanks, @owen2345 !

Changed in https://github.com/owen2345/camaleon-cms/pull/1085/commits/cad7c8e3e2a7623cafd42410045b61fe0907c8ac

One interesting observation is that some `jpeg` files were failing to load checking for the `script` with the previous implementation. With the current implementation they are not failing on `/<script[\s>]/i`, but on the `/on\w+\s*=/i` instead. Which make me wonder what is my iPhone adding to the photos??

Reply:
Also, I am thinking of implementing a Content Security Policy, which will allow scripts and other potentially unsafe things loaded only from the `assets/` path, making `media/` scripts unrenderable, WDYT?

Reply:
> Content Security Policy

 It makes sense, but are you thinking of creating another controller->action to check the security policy first?

Regarding "One interesting observation is that some...", can you change the `puts "Potentially..."` to print the part that matched the expression? Perhaps we need to improve the expression.

Reply:
The CSP is set in a Rails initializer and could be fine tuned there. But nope, it isn't at all reliable. Despite being version 3 already, the browsers are still struggling to support it. Especially with SVG, for example - https://github.com/mdn/content/pull/34592

Reply:
> Regarding "One interesting observation is that some...", can you change the `puts "Potentially..."` to print the part that matched the expression? Perhaps we need to improve the expression.

Yes, the matching part is `oNA=���`, see the screenshot from RubyMine:

<img width="1680" alt="Screenshot 2024-08-15 at 21 57 51" src="https://github.com/user-attachments/assets/09ed9806-1730-4bb2-a7a7-fd20bc145b18">